### ISSUE TITLE
Socket is not closed when handshake is failed

### ISSUE THREAD

**Issue by avasenin** [TakahikoKawasaki/nv-websocket-client#45]
We need to explicitly close the underlying socket when a handshake error happens. It could stay in ESTABLISHED or CLOSE_WAIT (depends on server implementation).

```
        try
        {
            // Perform WebSocket handshake.
            headers = shakeHands();
        }
        catch (WebSocketException e)
        {
            ( need to close underlying tcp Socket here)
            // Change the state to CLOSED.
            mStateManager.setState(CLOSED);

            // Notify the listener of the state change.
            mListenerManager.callOnStateChanged(CLOSED);

            // The handshake failed.
            throw e;
        }
```

**Comment by TakahikoKawasaki** [comment]
Released a new version `1.24`. Please try it.

**Comment by avasenin** [comment]
Cool! Yep, your patch works fine!

btw you could use your `closeSilently` method in https://github.com/TakahikoKawasaki/nv-websocket-client/blob/master/src/main/java/com/neovisionaries/ws/client/WebSocket.java#L3151
